### PR TITLE
Added some utilities written at ISIS

### PR DESCRIPTION
**Pull request by DominicOram**
This adds some utilities written at ISIS that could be helpful to the wider community.

I didn't think core.utils was a sensible place as they're more aimed at people who are actually building emulators.

### CODE REVIEW

File: lewis/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# lewis - a library for creating hardware device simulators
+# Copyright (C) 2016-2020 European Spallation Source ERIC
```
Comment (mattclarke) [inline lewis/utils/__init__.py]:
```suggestion
# Copyright (C) 2016-2021 European Spallation Source ERIC
```

**Comment by DominicOram** [comment]
Thanks @mattclarke, I'm fixing some of the flake8 errors now

File: lewis/utils/replies.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        - AttributeError if the first argument of the decorated function (self) does not contain .device or ._device
+        - AttributeError if the device does not contain a property called action
+
+    Usage:
+        @timed_reply(action="crash_pump", reply="WARNING: Input too quick", minimum_time_delay=150)
+        def acknowledge_pressure(channel):
+            return ACK
+    """
+    def decorator(func):
+        @six.wraps(func)
```
Comment (mattclarke) [inline lewis/utils/replies.py]: I guess this can be replaced with whatever the Python 3 equivalent is?

**Comment by mattclarke** [comment]
This looks useful: thanks @DominicOram and gang!

**Comment by DominicOram** [comment]
> This looks useful: thanks @DominicOram and gang!

Some of it's quite old, I suspect you may have written parts of it...

**Comment by DominicOram** [comment]
Think it's all done, happy for it to be merged whenever

**Comment by mattclarke** [comment]
Still seems to be some inconsistent docstring args.

**Comment by DominicOram** [comment]
Would be good to have this merged, you still looking at it @mattclarke or would you like me to take a look at the failing builds?

**Comment by mattclarke** [comment]
Still waiting for the docstrings args to be sorted out ;)

**Comment by DominicOram** [comment]
Sorry, missed that, look ok now?

**Comment by mattclarke** [comment]
Great, thank you.
I'll just take a quick look to see what is up with the build then I'll merge it

**Comment by mattclarke** [comment]
I'll fix the tests in a separate PR as it has nothing to do with this PR

**Pull request merged**